fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/DataConsumer.java

```diff
@@ -0,0 +1,20 @@
+package com.github.dedis.popstellar.utility.handler;
+
+import com.github.dedis.popstellar.model.network.method.message.data.Data;
+import com.github.dedis.popstellar.repository.LAORepository;
+import com.github.dedis.popstellar.utility.error.DataHandlingException;
+
+@FunctionalInterface
+public interface DataConsumer<T> {
+
+  void accept(
+      LAORepository laoRepository, String channel, T data, String messageId, String senderPk)
+      throws DataHandlingException;
```

Comment:
I feel like it would be great to have a function that takes only two arguments :
The data and a context that holds everything else.

This would make it a lot more maintainable for the future, as adding fields would not impact all handlers. And it will also make sure that a handler only use what it needs.

Reply:
Something like that for example ?
```
public final class HandlerContext {

  private final LAORepository laoRepository;
  private final String channel;
  private final MessageGeneral message;

  public HandlerContext(LAORepository laoRepository, String channel, MessageGeneral message) {
    this.laoRepository = laoRepository;
    this.channel = channel;
    this.message = message;
  }
  public LAORepository getLaoRepository() { return laoRepository; }
  public String getChannel() { return channel; }
  public MessageGeneral getMessage() { return message; }
  public String getMessageId() { return message.getMessageId(); }
  public String getSenderPk() { return message.getSender(); }
}
```
```
public static void handleCreateLao(HandlerContext context, CreateLao createLao) {
  Log.d(TAG, "handleCreateLao: channel " + context.getChannel() + ", msg=" + createLao);
  Lao lao = context.getLaoRepository().getLaoByChannel(context.getChannel());

  lao.setName(createLao.getName());
  lao.setCreation(createLao.getCreation());
  lao.setLastModified(createLao.getCreation());
  lao.setOrganizer(createLao.getOrganizer());
  lao.setId(createLao.getId());
  lao.setWitnesses(new HashSet<>(createLao.getWitnesses()));

  String publicKey = context.getLaoRepository().getPublicKey();
  if (lao.getOrganizer().equals(publicKey) || lao.getWitnesses().contains(publicKey)) {
    context.getLaoRepository().sendSubscribe(lao.getChannel() + "/consensus");
  }
  context.getLaoRepository().updateNodes(context.getChannel());
}
```